src/main/java/ubc/pavlab/rdp/services/UserGeneServiceImpl.java

```diff
@@ -158,30 +161,49 @@ private Set<UserGene> handleGeneSearchInternal( Gene gene, Set<TierType> tiers,
             organUberonIds = null;
         }
 
+        final Set<Integer> ontologyTermInfoIds;
+        if ( ontologyTermInfos != null ) {
+            ontologyTermInfoIds = ontologyService.inferTermIds( ontologyTermInfos );
+        } else {
+            ontologyTermInfoIds = null;
+        }
+
         // ortholog relationship is not reflexive (i.e. a gene is not its own ortholog), but we still want to display
         // that gene first when ortholog search is performed in the same MO
         if ( orthologTaxon == null || gene.getTaxon().equals( orthologTaxon ) ) {
             uGenes.addAll( userGeneRepository.findByGeneIdAndTierIn( gene.getGeneId(), tiers ).stream()
                     .filter( ug -> researcherPositions == null || researcherPositions.contains( ug.getUser().getProfile().getResearcherPosition() ) )
                     .filter( ug -> researcherCategories == null || containsAny( researcherCategories, ug.getUser().getProfile().getResearcherCategories() ) )
                     .filter( ortholog -> organUberonIds == null || containsAny( organUberonIds, ortholog.getUser().getUserOrgans().values().stream().map( UserOrgan::getUberonId ).collect( Collectors.toSet() ) ) )
+                    .filter( hasOntologyTermIn( ontologyTermInfoIds ) )
```

Comment:
This is temporary, it will eventually be switched to #190, which is why we're not changing the other filters.